cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataSubscriberService.java

```diff
@@ -279,9 +279,19 @@ public void processMessage(MetadataMessage message) {
 
     @Override
     public void processMessage(MetadataMessage message) {
-      FieldLineageInfo info = message.getPayload(GSON, FieldLineageInfo.class);
+      if (!(message.getEntityId() instanceof ProgramRunId)) {
+        LOG.warn("Missing program run id from the field lineage information. Ignoring the message {}", message);
```

Comment:
Ideally this should not happen since we always emit the message with the program run id. However I agree we should have metric in general about how many messages are being ignored by all TMS subscribers. Have file an improvement JIRA - https://issues.cask.co/browse/CDAP-13841 for fixing this across all subscriber/processors.